editor/editor_dock_manager.cpp

```diff
@@ -702,15 +702,22 @@ EditorDockManager::EditorDockManager() {
 	dock_select->set_v_size_flags(Control::SIZE_EXPAND_FILL);
 	dock_vb->add_child(dock_select);
 
-	if (!SceneTree::get_singleton()->get_root()->is_embedding_subwindows() && !EDITOR_GET("interface/editor/single_window_mode") && EDITOR_GET("interface/multi_window/enable")) {
-		dock_float = memnew(Button);
-		dock_float->set_text(TTR("Make Floating"));
-		dock_float->set_focus_mode(Control::FOCUS_NONE);
-		dock_float->set_h_size_flags(Control::SIZE_SHRINK_CENTER);
-		dock_float->connect("pressed", callable_mp(this, &EditorDockManager::_dock_make_selected_float));
-
-		dock_vb->add_child(dock_float);
+	dock_float = memnew(Button);
+	dock_float->set_text(TTR("Make Floating"));
+	if (!EditorNode::get_singleton()->is_multi_window_enabled()) {
+		dock_float->set_disabled(true);
+		if (SceneTree::get_singleton()->get_root()->is_embedding_subwindows()) {
+			dock_float->set_tooltip_text(TTR("Multi-window support is not available because the current platform doesn't support multiple windows."));
+		} else if (EDITOR_GET("interface/editor/single_window_mode")) {
+			dock_float->set_tooltip_text(TTR("Multi-window support is not available because Interface > Editor > Single Window Mode is enabled in the editor settings."));
+		} else if (EDITOR_GET("interface/multi_window/enable")) {
+			dock_float->set_tooltip_text(TTR("Multi-window support is not available because Interface > Multi Window > Enable is disabled in the editor settings."));
+		}
```

Comment:
This code should be moved to a method somewhere, like `get_multliwindow_support_tooltip()`.